tree.go

```diff
@@ -599,7 +599,7 @@ walk: // Outer loop for walking the tree
 		// Nothing found. We can recommend to redirect to the same URL with an
 		// extra trailing slash if a leaf exists for that path
 		value.tsr = path == "/" ||
-			(len(prefix) == len(path)+1 && n.handlers != nil)
+			(len(prefix) == len(path)+1 && path == prefix[:len(prefix) - 1] && n.handlers != nil)
```

Comment:
Can I restore the code in red as is?
![image](https://user-images.githubusercontent.com/67568757/131272823-6f73e531-ae7e-4ab3-afcc-46899ab2f4df.png)

Reply:
Yes, I run `go test` without any problems